api/modules/budgeting.js

```diff
@@ -1,9 +1,33 @@
 const moment = require('moment')
+const { DEFAULT_STRIPE_CURRENCY } = require('../config/constants')
 
 const budgeting = module.exports = (() => {
     const db = require('../models')
     const clientManagement = require('./clientManagement')
 
+    const checkExistingPaymentsWithExternalId = async (params) => {
```

Comment:
- Change to `checkForMismatchedClientPayments`
- Use `client_id` on the Payments to match `client.id` passed in from the function call